tests/msc3890_test.go

```diff
@@ -0,0 +1,87 @@
+//go:build msc3890
+// +build msc3890
+
+// This file contains tests for local notification settings as
+// defined by MSC3890, which you can read here:
+// https://github.com/matrix-org/matrix-doc/pull/3890
+
+package tests
+
+import (
+	"testing"
+	"time"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+	"github.com/tidwall/gjson"
+)
+
+func TestDeletingDeviceRemovesDeviceLocalNotificationSettings(t *testing.T) {
+	// Create a deployment with a single user
+	deployment := Deploy(t, b.BlueprintCleanHS)
+	defer deployment.Destroy(t)
+
+	// Create a user which we can log in to multiple times
+	alicePassword := "hunter2"
+	alice := deployment.RegisterUser(t, "hs1", "alice", alicePassword, false)
+
+	// Log in to another device on this user
+	_, secondDeviceAccessToken, secondDeviceID := alice.LoginUser(t, alicePassword)
+	secondDevice := &client.CSAPI{
+		AccessToken:      secondDeviceAccessToken,
+		BaseURL:          deployment.HS["hs1"].BaseURL,
+		Client:           client.NewLoggedClient(t, "hs1", nil),
+		SyncUntilTimeout: 5 * time.Second,
+		Debug:            alice.Debug,
+	}
```

Comment:
I'd be in favour of factoring this out into a method on `Deployment`, analogous to `RegisterUser`.
It's somewhat unexpected to use an already logged in client to log in again.

Reply:
Hmm, oddly I tried adding a `LoginUser` method initially in this PR and for whatever reason I was finding it difficult. But doing so again today in [cd70eb9](https://github.com/matrix-org/complement/pull/576/commits/cd70eb98d0ba71e3666a66358c1412a7c1ddb8d5) and it was a breeze and the test code is much cleaner!

So thanks for the suggestion :)